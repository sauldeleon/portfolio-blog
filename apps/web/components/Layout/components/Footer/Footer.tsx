import {
  CactusIcon,
  GithubIcon,
  LinkedinIcon,
  MoonIcon,
  TelegramIcon,
} from '@sdlgr/assets'
import { Footer as FooterLib } from '@sdlgr/footer'
import { mainTheme } from '@sdlgr/main-theme'

import { useClientTranslation } from '@web/i18n/client'

export function Footer() {
  const { t } = useClientTranslation('footer')

  return (
    <FooterLib
      tabIndex={-1}
      navProps={{ 'aria-label': t('siteMap') }}
      navItems={[
        { label: t('experience'), href: '/experience' },
        { label: t('contact'), href: '/contact' },
        { label: t('portfolio'), href: '/portfolio' },
        {
          label: t('darkMode'),
          href: '#',
          icon: (
            <MoonIcon color={mainTheme.colors.white} height={14} width={14} />
          ),
        },
        {
          label: t('painMode'),
          href: '#',
          icon: (
            <CactusIcon color={mainTheme.colors.white} height={14} width={14} />
          ),
        },
      ]}
      socialMediaItems={[
        {
          icon: <TelegramIcon color={mainTheme.colors.white} />,
          label: 'Telegram',
          href: 'https://t.me/Sdlgr',
        },
        {
          icon: <GithubIcon color={mainTheme.colors.white} />,
          label: 'Github',
          href: 'https://github.com/sauldeleon',
        },
        {
          icon: <LinkedinIcon color={mainTheme.colors.white} />,
          label: 'LinkedIn',
          href: 'https://www.linkedin.com/in/sauldeleonguerrero',
        },
      ]}
    />
  )
}
