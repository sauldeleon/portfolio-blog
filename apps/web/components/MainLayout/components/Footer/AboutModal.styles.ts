import Link from 'next/link'
import { Modal } from 'react-overlays'
import styled from 'styled-components'

import { CloseIcon } from '@sdlgr/assets'
import { Button } from '@sdlgr/button'
import { Heading, Label } from '@sdlgr/typography'

export const StyledBackdrop = styled.div`
  position: fixed;
  z-index: 1040;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: ${({ theme }) => theme.colors.black};
  opacity: 0.5;
`

export const StyledAboutModal = styled(Modal)`
  position: fixed;
  top: 30%;
  left: 50%;
  z-index: 1040;
  transform: translate(-51%, -40%);
  width: 100%;
  max-width: 800px;
  border: 1px solid ${({ theme }) => theme.colors.white};
  border-radius: 2px;
  background-color: ${({ theme }) => theme.colors.black};
  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
`

export const StyledModalContent = styled.div`
  position: relative;
  padding: 20px;
`

export const StyledModalHeading = styled(Heading)`
  margin-bottom: 40px;
`

export const StyledLabel = styled(Label)`
  margin: 20px 0 10px;
`

export const StyledModalCloseButton = styled(Button)`
  position: absolute;
  top: 20px;
  right: 20px;
  color: ${({ theme }) => theme.colors.white};
  ${({ theme }) => theme.helpers.textBottomBorder.removeAfter}
`

export const StyledPropertyWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  gap: 8px;
`

export const StyledCloseIcon = styled(CloseIcon)``

export const StyledIconLink = styled(Link)`
  ${({ theme }) => theme.helpers.textBottomBorder.removeAfter}
`

export const StyledList = styled.ul`
  list-style: none;
  display: flex;
  gap: 1rem;
`
