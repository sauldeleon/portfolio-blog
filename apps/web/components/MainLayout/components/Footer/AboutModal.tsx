import {
  FigmaIcon,
  GetTheIdeaIcon,
  GithubIcon,
  JestIcon,
  LinkedInIcon,
  NextJSIcon,
  NxIcon,
  StyledComponentsIcon,
  YarnIcon,
} from '@sdlgr/assets'
import { Body } from '@sdlgr/typography'

import { useClientTranslation } from '@web/i18n/client'

import {
  StyledIconLink,
  StyledLabel,
  StyledList,
  StyledModal,
  StyledModalHeading,
  StyledPropertyWrapper,
} from './AboutModal.styles'

interface AboutModalProps {
  isOpen: boolean
  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>
}

type IconListItemProps = {
  key: string
  href: string
  ariaLabel: string
  icon: React.ReactNode
}

export function AboutModal({ isOpen, setIsOpen }: AboutModalProps) {
  const { t } = useClientTranslation('footer')
  const developmentIcons: IconListItemProps[] = [
    {
      key: 'nextjs',
      href: 'https://nextjs.org/',
      ariaLabel: 'NextJS',
      icon: <NextJSIcon width={20} height={20} />,
    },
    {
      key: 'styled-components',
      href: 'https://styled-components.com/',
      ariaLabel: 'Styled-components',
      icon: <StyledComponentsIcon width={20} height={20} />,
    },
    {
      key: 'jest',
      href: 'https://jestjs.io/',
      ariaLabel: 'Jest',
      icon: <JestIcon width={20} height={20} />,
    },
    {
      key: 'nx',
      href: 'https://nx.dev/',
      ariaLabel: 'NX',
      icon: <NxIcon width={20} height={20} />,
    },
    {
      key: 'yarn',
      href: 'https://yarnpkg.com/',
      ariaLabel: 'Yarn',
      icon: <YarnIcon width={20} height={20} />,
    },
  ]

  const repositoryIcons: IconListItemProps[] = [
    {
      key: 'github',
      href: 'https://github.com/sauldeleon/portfolio-blog',
      ariaLabel: t('checkRepoHereAria'),
      icon: <GithubIcon width={20} />,
    },
  ]

  const designerIcons: IconListItemProps[] = [
    {
      key: 'linkedIn',
      href: 'https://www.linkedin.com/in/valentina-florentina-balta-cojocaru-stan-83619014a/',
      ariaLabel: t('checkLinkedInProfileAria', {
        name: 'Valentina Florentina Balta-Cojocaru-Stan',
      }),
      icon: <LinkedInIcon width={20} height={20} />,
    },
    {
      key: 'getTheIdea',
      href: 'http://valentina.pro/index.html',
      ariaLabel: t('getTheIdeaAria', {
        name: 'Valentina Florentina Balta-Cojocaru-Stan',
      }),
      icon: <GetTheIdeaIcon width={20} />,
    },
  ]

  const designIcons: IconListItemProps[] = [
    {
      key: 'figma',
      href: 'https://www.figma.com/file/ZgfaiU473XMQAWz9kbiMO6/Saul-portfolio-%2B-design-library',
      ariaLabel: t('checkDesignHereAria'),
      icon: <FigmaIcon width={20} height={20} />,
    },
  ]

  const modalSections = [
    {
      label: t('development'),
      key: 'development',
      properties: [
        { label: t('usedTechs'), key: 'usedTechs', icons: developmentIcons },
        { label: t('repository'), key: 'repository', icons: repositoryIcons },
      ],
    },
    {
      label: t('design'),
      key: 'design',
      properties: [
        { label: t('designer'), key: 'designer', icons: designerIcons },
        { label: t('designLink'), key: 'designLink', icons: designIcons },
      ],
    },
  ]

  return (
    <StyledModal isOpen={isOpen} setIsOpen={setIsOpen}>
      <section>
        <StyledModalHeading $level={2} as="h4" id="about-modal-label">
          {t('aboutModalTitle')}
        </StyledModalHeading>

        {modalSections.map(({ label, key, properties }) => (
          <div key={key}>
            <StyledLabel $level="L">{label}</StyledLabel>
            {properties.map(({ label, key, icons }) => (
              <StyledPropertyWrapper key={key}>
                <Body>{label}</Body>
                <StyledList>
                  {icons.map(({ key, href, icon, ariaLabel }) => (
                    <li key={key}>
                      <StyledIconLink aria-label={ariaLabel} href={href}>
                        {icon}
                      </StyledIconLink>
                    </li>
                  ))}
                </StyledList>
              </StyledPropertyWrapper>
            ))}
          </div>
        ))}
      </section>
    </StyledModal>
  )
}
