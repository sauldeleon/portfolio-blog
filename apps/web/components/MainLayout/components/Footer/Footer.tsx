import { usePathname, useRouter } from 'next/navigation'
import { useContext } from 'react'

import {
  CactusIcon,
  GithubIcon,
  LanguageIcon,
  LinkedinIcon,
  MoonIcon,
  TelegramIcon,
} from '@sdlgr/assets'
import { Footer as FooterLib, NavItem, SocialMediaItem } from '@sdlgr/footer'
import { LanguageContext } from '@sdlgr/i18n-config'
import { mainTheme } from '@sdlgr/main-theme'

import { useClientTranslation } from '@web/i18n/client'

export function Footer() {
  const { t } = useClientTranslation('footer')
  const { language } = useContext(LanguageContext)
  const pathname = usePathname()
  const { push } = useRouter()

  const toggleLanguage = (path: string) => {
    const noLangPath = path.replace(/^\/[\w\d]+/, '')
    const newLanguage = language === 'en' ? 'es' : 'en'
    push(`/${newLanguage}${noLangPath}`)
  }

  const navItems: NavItem[] = [
    {
      label: t('experience'),
      ariaLabel: t('experienceAria'),
      href: `/${language}/experience`,
    },
    {
      label: t('contact'),
      ariaLabel: t('contactAria'),
      href: `/${language}/contact`,
    },
    {
      label: t('portfolio'),
      ariaLabel: t('portfolioAria'),
      href: `/${language}/portfolio`,
    },
    {
      label: t('darkMode'),
      ariaLabel: t('darkModeAria'),
      onClick: (e) => {
        e.preventDefault()
        console.log('dark mode')
      },
      icon: <MoonIcon color={mainTheme.colors.white} height={14} width={14} />,
    },
    {
      label: t('painMode'),
      ariaLabel: t('painModeAria'),
      onClick: (e) => {
        e.preventDefault()
        console.log('pain mode')
      },
      icon: (
        <CactusIcon color={mainTheme.colors.white} height={14} width={14} />
      ),
    },
    {
      label: language === 'en' ? 'EN' : 'ES',
      ariaLabel: t('toggleLanguage', {
        language: language === 'en' ? 'English' : 'EspaÃ±ol',
      }),
      onClick: () => toggleLanguage(pathname),
      icon: (
        <LanguageIcon color={mainTheme.colors.white} height={22} width={22} />
      ),
    },
  ]

  const socialMediaItems: SocialMediaItem[] = [
    {
      icon: <TelegramIcon color={mainTheme.colors.white} />,
      ariaLabel: t('telegramAria'),
      href: 'https://t.me/Sdlgr',
    },
    {
      icon: <GithubIcon color={mainTheme.colors.white} />,
      ariaLabel: t('githubAria'),
      href: 'https://github.com/sauldeleon',
    },
    {
      icon: <LinkedinIcon color={mainTheme.colors.white} />,
      ariaLabel: t('linkedInAria'),
      href: 'https://www.linkedin.com/in/sauldeleonguerrero',
    },
  ]

  return (
    <FooterLib
      tabIndex={-1}
      navProps={{ 'aria-label': t('siteMap') }}
      navItems={navItems}
      socialMediaItems={socialMediaItems}
    />
  )
}
