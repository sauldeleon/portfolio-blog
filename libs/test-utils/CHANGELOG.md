# Changelog Test utils

## [0.2.3](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.2...test-utils-0.2.3) (2024-01-08)

## [0.2.2](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.1...test-utils-0.2.2) (2023-09-26)

## [0.2.1](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.0...test-utils-0.2.1) (2023-09-14)

# [0.2.0](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.1.0...test-utils-0.2.0) (2023-09-05)

### Bug Fixes

- reduced duplicated lines ([f51168a](https://github.com/sauldeleon/portfolio-blog/commit/f51168aefe025a5a65ffdb2dc22f7a16b863406e))
- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))

# 0.1.0 (2023-09-02)

### Bug Fixes

- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- add contact initial page ([fea862b](https://github.com/sauldeleon/portfolio-blog/commit/fea862b3507268013149050ae3f05fa517e7b3d9))
- add homepage ([60523aa](https://github.com/sauldeleon/portfolio-blog/commit/60523aa31ec088d318745ef0c757b4cbe656f0d4))
- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))
- improved a11y and styles ([873f656](https://github.com/sauldeleon/portfolio-blog/commit/873f656587f50df8e29ad526fdc68fea0b2c9784))
- toggle language button ([afd97e9](https://github.com/sauldeleon/portfolio-blog/commit/afd97e95beac0f70ed643c753e71db2ca014054b))
