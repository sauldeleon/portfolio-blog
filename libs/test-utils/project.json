{
  "name": "test-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/test-utils/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "version": {
      "executor": "@jscutlery/semver:version",
      "options": {
        "changelogHeader": "# Changelog Test utils \n",
        "baseBranch": "main",
        "commitMessageFormat": "chore(${projectName}): release: v${version}",
        "push": false
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/test-utils/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/test-utils/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
