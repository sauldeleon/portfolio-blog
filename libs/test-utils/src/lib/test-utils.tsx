import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { Matcher } from '@testing-library/dom/types/matches'
import {
  RenderOptions,
  RenderResult,
  queryAllByAttribute as _queryAllByAttribute,
  queryByAttribute as _queryByAttribute,
  act,
  render,
  screen,
} from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'
import Router, { NextRouter } from 'next/router'
import { ReactNode, useEffect, useState } from 'react'
import { ThemeProvider } from 'styled-components'

import { LanguageContextProvider } from '@sdlgr/i18n-config'
import { mainTheme } from '@sdlgr/main-theme'

export type MatcherCustomFnParams = {
  content: string
  element: Element | null
}

/* istanbul ignore next */
export const renderWithTheme = (children: ReactNode): RenderResult => {
  return render(<ThemeProvider theme={mainTheme}>{children}</ThemeProvider>)
}

/* istanbul ignore next */
export function RenderProviders({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

  return (
    <QueryClientProvider client={queryClient}>
      <LanguageContextProvider value={{ language: 'en' }}>
        <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
      </LanguageContextProvider>
    </QueryClientProvider>
  )
}

/* istanbul ignore next */
export const renderApp = (
  children: ReactNode,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(<RenderProviders>{children}</RenderProviders>, options)
}

/**
 * Generic function to get element by attribute and value
 * @example queryByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function queryByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example queryByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function queryAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Generic function to get element by attribute and value
 * @example getByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function getByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example getByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function getAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Shortcut version for queryByAttribute, but specific for id
 * @param {HTMLElement} container
 * @param {string} id
 * @return {any}
 */
export function getById(container: HTMLElement, id: string) {
  return queryByAttribute('id')(container, id) as Element
}

/**
 * Shortcut version for queryByAttribute, but specific for class name
 * @param {HTMLElement} container
 * @param {string} className
 * @return {any}
 */
export function getByClassName(container: HTMLElement, className: string) {
  return queryByAttribute('class')(container, className) as Element
}

function matcherCustomFn(
  tag: string,
  matcher: Matcher,
  { element, content }: MatcherCustomFnParams
) {
  const isTag = element?.tagName.toUpperCase() === tag.toUpperCase()
  let matching = false
  if (isTag) {
    matching = content === matcher
    /* istanbul ignore else*/
    if (typeof matcher === 'function') {
      matching = matcher(content, element)
    } else if (matcher instanceof RegExp) {
      matching = matcher.test(content)
    }
  }
  return matching && isTag
}

/**
 * Get 1 element by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getByTagAndText(tag: string, matcher: Matcher) {
  return screen.getByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Get multiple elements by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getAllByTagAndText(tag: string, matcher: Matcher) {
  return screen.getAllByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Mock Math.random() to return a specific value
 * @param {number} value
 */
export function mockRandom(value = 0.1234) {
  const mockMath = Object.create(global.Math)
  mockMath.random = () => value
  global.Math = mockMath
}

/**
 * Hook to use in test to ensure "last" render is called
 */
export function useWithRefreshInTest() {
  const [withRefreshInTest, setWithRefreshInTest] = useState(false)

  useEffect(() => {
    // Change state to "rerender" the component
    setWithRefreshInTest(true)
  }, [])

  return withRefreshInTest
}

export { render } from '@testing-library/react'

/* istanbul ignore next */
export const awaitAllAsync = async () =>
  act(() => new Promise((resolve) => setTimeout(resolve, 0)))

/* istanbul ignore next */
export function renderWithNextRouter(
  children: ReactNode,
  options?: Partial<NextRouter>
) {
  return renderWithTheme(
    <RouterContext.Provider value={{ ...Router, ...options }}>
      {children}
    </RouterContext.Provider>
  )
}
