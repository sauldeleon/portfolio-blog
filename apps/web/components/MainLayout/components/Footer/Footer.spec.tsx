import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { usePathname } from 'next/navigation'

import { renderApp } from '@sdlgr/test-utils'

import { Footer } from './Footer'

const mockPush = jest.fn()
jest.mock('next/navigation', () => ({
  usePathname: jest.fn(),
  useRouter: () => ({
    push: mockPush,
  }),
}))

describe('Footer', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  it('should render successfully', async () => {
    const { baseElement } = renderApp(<Footer />)
    await screen.findByRole('navigation')
    expect(screen.getAllByRole('link')).toHaveLength(6)
    expect(screen.getAllByRole('button')).toHaveLength(3)
    expect(baseElement).toMatchSnapshot()
  })

  it('should trigger actions in buttons successfully', async () => {
    jest.spyOn(console, 'log')
    const mockFn = jest.fn()
    ;(console.log as jest.Mock).mockImplementation(mockFn)

    renderApp(<Footer />)
    await screen.findByRole('navigation')
    await userEvent.click(screen.getByRole('button', { name: /Dark mode/ }))
    await userEvent.click(screen.getByRole('button', { name: /Pain mode/ }))
    expect(mockFn).toHaveBeenCalledTimes(2)
  })

  it('should toggle language to Spanish', async () => {
    ;(usePathname as jest.Mock).mockImplementation(() => '/en/path/slug')
    renderApp(<Footer />)
    await screen.findByRole('navigation')
    const toggleLanguageButton = screen.getByRole('button', {
      name: /Toggle language/,
    })
    await userEvent.click(toggleLanguageButton)
    expect(mockPush).toHaveBeenNthCalledWith(1, '/es/path/slug')
  })

  it('should toggle language to English', async () => {
    ;(usePathname as jest.Mock).mockImplementation(() => '/es/path/slug')
    renderApp(<Footer />, undefined, { language: 'es' })
    await screen.findByRole('navigation')
    const toggleLanguageButton = screen.getByRole('button', {
      name: /Toggle language/,
    })
    await userEvent.click(toggleLanguageButton)
    expect(mockPush).toHaveBeenNthCalledWith(1, '/en/path/slug')
  })
})
