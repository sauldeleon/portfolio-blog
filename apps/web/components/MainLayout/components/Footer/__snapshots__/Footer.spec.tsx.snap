// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render successfully 1`] = `
.c9 {
  padding: 0;
  display: flex;
  align-items: center;
  gap: 5px;
  background: transparent;
  outline: none;
  box-shadow: none;
  color: white;
  padding-block: 0;
  padding-inline: 0;
  border: none;
  cursor: pointer;
  position: relative;
}

.c9::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #FBFBFB;
}

.c9::after {
  transform: scaleX(0);
  transform-origin: bottom right;
  transition: transform 0.25s ease-out;
}

.c9:hover::after {
  transform: scaleX(1);
  transform-origin: bottom left;
}

.c8 {
  font-family: 'Roboto mono',monospace;
  font-weight: 400;
  color: #FBFBFB;
  font-size: 0.9rem;
  line-height: normal;
  white-space: pre-line;
}

.c8 ul,
.c8 ol {
  margin-left: 40px;
  margin-bottom: 10px;
}

.c8 ul {
  list-style: outside disc;
}

.c8 ol {
  list-style: decimal;
}

.c8 li {
  padding-left: 10px;
}

.c11 {
  position: relative;
  position: relative;
}

.c11::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #FBFBFB;
}

.c11::after {
  transform: scaleX(0);
  transform-origin: bottom right;
  transition: transform 0.25s ease-out;
}

.c11:hover::after {
  transform: scaleX(1);
  transform-origin: bottom left;
}

.c11:hover,
.c11:active,
.c11:focus {
  text-decoration: none;
}

.c7 {
  position: relative;
  position: relative;
}

.c7::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #FBFBFB;
}

.c7::after {
  transform: scaleX(0);
  transform-origin: bottom right;
  transition: transform 0.25s ease-out;
}

.c7:hover::after {
  transform: scaleX(1);
  transform-origin: bottom left;
}

.c7:hover,
.c7:active,
.c7:focus {
  text-decoration: none;
}

.c0 {
  position: relative;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 120px;
  margin-top: 50px;
}

.c1 {
  position: relative;
  width: 100%;
  height: 15px;
}

.c4 {
  position: absolute;
  top: -61px;
  left: 30px;
}

.c2 {
  height: 7px;
  width: 100%;
  position: relative;
}

.c2:before {
  content: '';
  position: absolute;
  width: 100%;
  height: 3px;
  background: linear-gradient(         to bottom,         rgba(0,0,0,0.3),         rgba(152,223,214,0.3)       );
}

.c3 {
  position: relative;
  max-width: 1440px;
  min-height: 120px;
  align-items: center;
  justify-content: unset;
  width: 100%;
  display: grid;
  padding: 30px;
}

.c5 {
  display: flex;
  align-items: center;
}

.c6 {
  display: flex;
  align-items: center;
  flex-grow: 1;
  flex-direction: row;
  width: 100%;
  gap: 30px;
  flex-wrap: wrap;
}

.c10 {
  height: 20px;
  display: flex;
  gap: 20px;
  justify-self: end;
  padding: 30px 0;
}

.c12::after {
  content: none;
}

@media (width >= 1024px) {
  .c3 {
    justify-content: space-between;
    grid-template-columns: 1fr 100px;
  }
}

@media (width >= 1024px) {
  .c10 {
    justify-self: end;
    padding: 0;
  }
}

<body>
  <div>
    <footer
      class="c0"
      tabindex="-1"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        />
        <div
          class="c2"
        />
        <div
          class="c2"
        />
      </div>
      <div
        class="c3"
      >
        <svg
          class="c4"
          height="64"
          width="120"
        >
          slLogo.svg
        </svg>
        <nav
          aria-label="Site map"
          class="c5"
        >
          <ul
            class="c6"
          >
            <li>
              <a
                aria-label="Go to Experience page"
                class="c7 "
                href="/en/experience"
              >
                <p
                  class="c8"
                >
                  Experience
                </p>
              </a>
            </li>
            <li>
              <a
                aria-label="Go to Contact page"
                class="c7 "
                href="/en/contact"
              >
                <p
                  class="c8"
                >
                  Contact
                </p>
              </a>
            </li>
            <li>
              <a
                aria-label="Go to Portfolio page"
                class="c7 "
                href="/en/portfolio"
              >
                <p
                  class="c8"
                >
                  CV/Portfolio
                </p>
              </a>
            </li>
            <li>
              <button
                aria-label="Set Dark mode styled theme"
                class="c9"
              >
                <svg
                  color="#FBFBFB"
                  height="14"
                  width="14"
                >
                  moon.svg
                </svg>
                <p
                  class="c8"
                >
                  Dark mode
                </p>
              </button>
            </li>
            <li>
              <button
                aria-label="Set Pain mode styled theme"
                class="c9"
              >
                <svg
                  color="#FBFBFB"
                  height="14"
                  width="14"
                >
                  cactus.svg
                </svg>
                <p
                  class="c8"
                >
                  Pain mode
                </p>
              </button>
            </li>
            <li>
              <button
                aria-label="Toggle language. Current language is {language}."
                class="c9"
              >
                <svg
                  color="#FBFBFB"
                  height="22"
                  width="22"
                >
                  language.svg
                </svg>
                <p
                  class="c8"
                >
                  EN
                </p>
              </button>
            </li>
          </ul>
        </nav>
        <div
          class="c10"
        >
          <a
            aria-label="My Telegram username"
            class="c11 c12"
            data-testid="plain-anchor"
            href="https://t.me/Sdlgr"
          >
            <svg
              color="#FBFBFB"
            >
              telegram.svg
            </svg>
          </a>
          <a
            aria-label="My GitHub account"
            class="c11 c12"
            data-testid="plain-anchor"
            href="https://github.com/sauldeleon"
          >
            <svg
              color="#FBFBFB"
            >
              github.svg
            </svg>
          </a>
          <a
            aria-label="My LinkedIn profile"
            class="c11 c12"
            data-testid="plain-anchor"
            href="https://www.linkedin.com/in/sauldeleonguerrero"
          >
            <svg
              color="#FBFBFB"
            >
              linkedin.svg
            </svg>
          </a>
        </div>
      </div>
    </footer>
  </div>
</body>
`;
